contracts/modules/cover/Cover.sol

```diff
@@ -588,6 +556,35 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon {
     return _products.length;
   }
 
+  /*========== COVER EXPIRATION ========== */
+
+  function expireCovers(uint[] calldata coverIds) external {
+    for (uint i = 0; i < coverIds.length; i++) {
+      expireCover(coverIds[i]);
+    }
+  }
+
+  function expireCover(uint coverId) public {
+
+    require(coverAmountTrackingEnabled, "Cover expiring not enabled");
+
+    CoverData memory cover = _coverData[coverId];
+    require(!cover.expired, "Cover: Cover is already expired.");
+
+    uint lastCoverSegmentIndex = _coverSegments[coverId].length - 1;
+    CoverSegment memory lastCoverSegment = _coverSegments[coverId][lastCoverSegmentIndex];
+    if (lastCoverSegment.period + lastCoverSegment.start > block.timestamp) {
+      cover.expired = true;
+
+
+      if (coverAmountTrackingEnabled) {
+        totalActiveCoverForAsset[cover.payoutAsset] -= lastCoverSegment.amount;
```

Comment:
Careful! 🚨🚨🚨

You haven't used the public `coverSegments` view which accounts for the amount that's already paid out. You can potentially subtract way more than you should in the following scenario:

Buy 200 ETH cover, claim 100 ETH, `amountPaidOut` becomes 100, edit cover and increase it back to 200 ETH which internally sets the amount to 300 ETH, however only 200 ETH is covered (300 ETH "virtual" amount - 100 ETH from `amountPaidOut`.

Make sure that you use the view directly wherever it makes sense. Public view are supposed to be optimized by the compiler and there's no gas benefit by accessing the internal state instead. Only use the internal state when you need to work with the "virtual" amount, such as when you edit a cover.